### PR TITLE
Improve ci

### LINKED ISSUE #5: Configure CI/CD
Would be great have the CI/CD configured on CircleCI or maybe GitHub Actions.

### PR DESCRIPTION
**Pull request by thiamsantos**
Closes #5 

Steps:

- Format
- Credo
- Tests
- Check version
  -- If branch != master
- Publish
  -- if branch == master
- Github Release
  -- if branch == master

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-elixir@v1.0.0
+        with:
+          otp-version: "22.3"
+          elixir-version: "1.10.2"
+      - name: Install Dependencies
+        run: mix deps.get
+      - name: Run Credo
+        run: mix credo
```
Comment (brosquinha) [inline .github/workflows/ci.yml:34]: Why not?
```suggestion
        run: mix credo --strict
```

Comment (thiamsantos) [reply .github/workflows/ci.yml:34]: @brosquinha It is already setted in the config file :)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,9 +55,11 @@ jobs:
         run: mix deps.get
       - name: Run Tests
         run: mix test
-  mix_format:
-    name: mix format (Elixir 1.10.2 OTP 22.3)
+  mix_check_version:
+    name: Check version (Elixir 1.10.2 OTP 22.3)
     runs-on: ubuntu-16.04
+    needs: [mix_format, mix_credo, mix_test]
+    if: github.ref != 'refs/heads/master'
```
Comment (Danielwsx64) [inline .github/workflows/ci.yml:62]: WDT about:

```suggestion
    if: github.ref == 'refs/heads/master' && github.event == 'pull_request'
```
Run just for PRs to master branch

File: .credo.exs [hunk 3]
```diff
@@ -33,7 +37,11 @@
       # If you want to enforce a style guide and need a more traditional linting
       # experience, you can change `strict` to `true` below:
       #
-      strict: false,
+      strict: true,
```
Comment (RafaelCamarda) [inline .credo.exs:40]: :heart:

File: .github/workflows/ci.yml (hunk 2 above)
Comment (thiamsantos) [reply .github/workflows/ci.yml]: I hope that the changes will work @Danielwsx64, now the check version job will only run for branches different than master, when the event is pull request and the branch to merge is master. So only PRs to master. `github.ref != 'refs/heads/master' && github.event_name == 'pull_request' && github.base_ref == 'master'`

WDYT?

Comment (Danielwsx64) [reply .github/workflows/ci.yml]: perfect :+1:

**Comment by thiamsantos** [comment]
@Klockner @brosquinha @RafaelCamarda @Danielwsx64 https://github.com/brainn-co/xcribe/releases/tag/v0.6.1 The creation of the release and tag worked great.

**Pull request merged**